### DISCUSSION TITLE
Running a Hybrid Dagster+ agent in a Docker container on ACI

### DISCUSSION THREAD

**Discussion by bosmat-elementl** [dagster-io/dagster#16286] [Q&A]
Can I setup a hybrid agent running in a docker container on ACI?

**Comment by bosmat-elementl** [comment]
I would expect the docker agent to work here running in an ACI machine, but we don't yet have a deeper ACI integration for dagster cloud the same way that we do for Amazon ECS

**Reply by hwspuijbroek** [reply]
@bosmat-elementl did you receive any update on this? I am interested in this also.

**Reply by garethbrickman** [reply]
We don't have a formal ACI integration in Dagster+ but you should be able to manually configure the [Hybrid Docker agent](https://docs.dagster.io/dagster-plus/deployment/deployment-types/hybrid/docker/setup) in Azure.

**Reply by hwspuijbroek** [reply]
@garethbrickman I thought that the Dagster+ needs the proper user_code_launcher to work with Dagster in ACI.

I tried multiple modules (K8), but it wasn't working. Don't know what to use/to be expect. So you would expect that the code below would work;

user_code_launcher:
  module: dagster_cloud.workspace.docker
  class: DockerUserCodeLauncher

**Reply by garethbrickman** [reply]
I can't be certain and it's entirely possible ACI isn't compatible but other users have been able to run the Docker agent in Azure but perhaps using another service. Developing a dedicated Azure agent and natively supporting ACI is in the product backlog.
